sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/container/ThroughputContainerController.java

```diff
@@ -0,0 +1,321 @@
+    private final AtomicReference<Integer> maxContainerThroughput;
+    private final RxPartitionKeyRangeCache partitionKeyRangeCache;
+    private final CosmosAsyncContainer targetContainer;
+
+    private final CancellationTokenSource cancellationTokenSource;
+    private final Scheduler scheduler;
+
+    private ThroughputGroupControllerBase defaultGroupController;
+    private String targetContainerRid;
+    private String targetDatabaseRid;
+    private ThroughputResolveLevel throughputResolveLevel;
+
+    public ThroughputContainerController(
+        ConnectionMode connectionMode,
+        GlobalEndpointManager globalEndpointManager,
+        List<ThroughputControlGroup> groups,
+        RxPartitionKeyRangeCache partitionKeyRangeCache) {
+
+        checkNotNull(globalEndpointManager, "GlobalEndpointManager can not be null");
+        checkArgument(groups != null && groups.size() > 0, "Throughput budget groups can not be null or empty");
+        checkNotNull(partitionKeyRangeCache, "RxPartitionKeyRangeCache can not be null");
+
+        this.connectionMode = connectionMode;
+        this.globalEndpointManager = globalEndpointManager;
+        this.groupControllers = new ConcurrentHashMap<>();
+        this.groups = groups;
+        this.maxContainerThroughput = new AtomicReference<>();
+        this.partitionKeyRangeCache = partitionKeyRangeCache;
+
+        this.targetContainer = groups.get(0).getTargetContainer();
+        this.client = CosmosBridgeInternal.getContextClient(this.targetContainer);
+
+        this.throughputResolveLevel = this.getThroughputResolveLevel(groups);
+
+        this.cancellationTokenSource = new CancellationTokenSource();
+        this.scheduler = Schedulers.elastic();
+    }
+
+    private ThroughputResolveLevel getThroughputResolveLevel(List<ThroughputControlGroup> groupConfigs) {
+        if (groupConfigs.stream().anyMatch(groupConfig -> groupConfig.getTargetThroughputThreshold() != null)) {
+            // Throughput can be provisioned on container level or database level, will start from container
+            return ThroughputResolveLevel.CONTAINER;
+        } else {
+            // There is no group configured with throughput threshold, so no need to query throughput
+            return ThroughputResolveLevel.NONE;
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Mono<T> init() {
+        return this.resolveDatabaseResourceId()
+            .flatMap(controller -> this.resolveContainerResourceId())
+            .flatMap(controller -> this.resolveContainerMaxThroughput())
+            .flatMap(controller -> this.createAndInitializeGroupControllers())
+            .doOnSuccess(controller -> {
+                this.setDefaultGroupController();
+                scheduler.schedule(() -> this.refreshContainerMaxThroughputTask(this.cancellationTokenSource.getToken()).subscribe());
+            })
+            .thenReturn((T) this);
```

Comment:
no error handler here ?

Reply:
nope, the error will bubble up to ThroughputControlStore, there is no specific error logic handling here.